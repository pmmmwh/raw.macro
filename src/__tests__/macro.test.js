const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");

pluginTester({
  babelOptions: { filename: __filename },
  plugin,
  tests: {
    "no usage": {
      code: `import raw from '../macro'`,
      snapshot: true,
    },
    "correct usage": {
      code: `
        import raw from '../macro';

        const macro = raw('raw.macro');
        const md = raw('./fixtures/markdown.md');
        const js = raw('./fixtures/javascript.js');
      `,
      snapshot: true,
    },
    "static template literal": {
      code: `
        import raw from '../macro';

        const macro = raw('raw.macro');
        const md = raw(\`./fixtures/markdown.md\`);
        const js = raw(\`./fixtures/javascript.js\`);
      `,
      snapshot: true,
    },
    "static template literal using interpolation": {
      code: `
        import raw from '../macro';

        const fileName = 'markdown';
        const md = raw(\`./fixtures/\${fileName}.md\`);
      `,
      snapshot: true,
    },
    "dynamic import directory": {
      code: `
        import raw from '../macro';

        function dynamic(locale) {
          const md = raw(\`./fixtures/\${locale}/post.md\`);
          return md;
        }

        const data = dynamic('en')
      `,
      snapshot: true,
    },
    "dynamic import file name": {
      code: `
        import raw from '../macro';

        function dynamic(fileName) {
          const md = raw(\`./fixtures/\${fileName}.md\`);
          return md;
        }

        const data = dynamic('markdown')
      `,
      snapshot: true,
    },
    "dynamic import multiple variable": {
      code: `
        import raw from '../macro';

        const md = raw(\`./fixtures/\${locale}/\${fileName}.md\`);
      `,
      snapshot: true,
    },
    "multiple dynamic import": {
      code: `
        import raw from '../macro';

        const a0 = raw('./fixtures/markdown.md');
        const a1 = raw(\`./\${fixtureDir}/markdown.md\`);
        const a2 = raw(\`./fixtures/\${fileName}\`);
      `,
      snapshot: true,
    },
    "custom encoding": {
      code: `
        import raw from '../macro';

        const a0 = raw('./fixtures/markdown.md', 'utf-8');
        const a1 = raw(\`./\${fixtureDir}/markdown.md\`, 'binary');
        const a2 = raw(\`./fixtures/\${fileName}\`, 'binary');
      `,
      snapshot: true,
    },
    "invalid file in dynamic directory": {
      code: `
        import raw from '../macro';

        const a1 = raw(\`./\${fixtureDir}/index.js\`);
      `,
      throws: /Cannot resolve file (.+) in these directories/,
    },
    "invalid dynamic value at the start of template literal": {
      code: `
        import raw from '../macro';

        const a1 = raw(\`\${fixtureDir}/javascript.js\`);
      `,
      throws:
        "Invalid value, variable interpolation can't be at the start of template literal",
    },
    "invalid dynamic values exceed limit": {
      code: `
        import raw from '../macro';

        const a1 = raw(\`./\${anotherDir}/\${fixtureDir}/\${fileName}.md\`);
      `,
      throws:
        "Invalid value. You can only have 2 dynamic values max. 1 for directory name, 1 for file name.",
    },
  },
});
