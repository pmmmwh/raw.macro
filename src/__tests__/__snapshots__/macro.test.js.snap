// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. no usage: 1. no usage 1`] = `

import raw from '../macro'

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros 2. correct usage: 2. correct usage 1`] = `


import raw from '../macro';

const macro = raw('raw.macro');
const md = raw('./fixtures/markdown.md');
const js = raw('./fixtures/javascript.js');
      

      ↓ ↓ ↓ ↓ ↓ ↓

const macro =
  '/* istanbul ignore next */\\nimport macro from "./macro";\\n\\nexport default macro;\\n';
const md = "# This is the title\\n\\nThis is the description\\n";
const js =
  'const sum = require("../sum");\\n\\ntest("add two numbers", () => {\\n  const result = sum(1, 2);\\n  expect(result).toEqual(3);\\n});\\n';

`;

exports[`macros 3. static template literal: 3. static template literal 1`] = `


import raw from '../macro';

const macro = raw('raw.macro');
const md = raw(\`./fixtures/markdown.md\`);
const js = raw(\`./fixtures/javascript.js\`);
      

      ↓ ↓ ↓ ↓ ↓ ↓

const macro =
  '/* istanbul ignore next */\\nimport macro from "./macro";\\n\\nexport default macro;\\n';
const md = "# This is the title\\n\\nThis is the description\\n";
const js =
  'const sum = require("../sum");\\n\\ntest("add two numbers", () => {\\n  const result = sum(1, 2);\\n  expect(result).toEqual(3);\\n});\\n';

`;

exports[`macros 4. static template literal using interpolation: 4. static template literal using interpolation 1`] = `


import raw from '../macro';

const fileName = 'markdown';
const md = raw(\`./fixtures/\${fileName}.md\`);
      

      ↓ ↓ ↓ ↓ ↓ ↓

const fileName = "markdown";
const md = "# This is the title\\n\\nThis is the description\\n";

`;

exports[`macros 5. dynamic import directory: 5. dynamic import directory 1`] = `


import raw from '../macro';

function dynamic(locale) {
  const md = raw(\`./fixtures/\${locale}/post.md\`);
  return md;
}

const data = dynamic('en')
      

      ↓ ↓ ↓ ↓ ↓ ↓

var __raw_dynamic__0 = {
  en: "English\\n",
  id: "Bahasa\\n",
};
function dynamic(locale) {
  const md = __raw_dynamic__0[locale];
  return md;
}
const data = dynamic("en");

`;

exports[`macros 6. dynamic import file name: 6. dynamic import file name 1`] = `


import raw from '../macro';

function dynamic(fileName) {
  const md = raw(\`./fixtures/\${fileName}.md\`);
  return md;
}

const data = dynamic('markdown')
      

      ↓ ↓ ↓ ↓ ↓ ↓

var __raw_dynamic__0 = {
  mark: "## Another markdown\\n",
  markdown: "# This is the title\\n\\nThis is the description\\n",
};
function dynamic(fileName) {
  const md = __raw_dynamic__0[fileName];
  return md;
}
const data = dynamic("markdown");

`;

exports[`macros 7. dynamic import multiple variable: 7. dynamic import multiple variable 1`] = `


import raw from '../macro';

const md = raw(\`./fixtures/\${locale}/\${fileName}.md\`);
      

      ↓ ↓ ↓ ↓ ↓ ↓

var __raw_dynamic__0 = {
  en: {
    post: "English\\n",
  },
  id: {
    post: "Bahasa\\n",
  },
};
const md = __raw_dynamic__0[locale][fileName];

`;

exports[`macros 8. multiple dynamic import: 8. multiple dynamic import 1`] = `


import raw from '../macro';

const a0 = raw('./fixtures/markdown.md');
const a1 = raw(\`./\${fixtureDir}/markdown.md\`);
const a2 = raw(\`./fixtures/\${fileName}\`);
      

      ↓ ↓ ↓ ↓ ↓ ↓

var __raw_dynamic__1 = {
  "javascript.js":
    'const sum = require("../sum");\\n\\ntest("add two numbers", () => {\\n  const result = sum(1, 2);\\n  expect(result).toEqual(3);\\n});\\n',
  "mark.md": "## Another markdown\\n",
  "markdown.md": "# This is the title\\n\\nThis is the description\\n",
};
var __raw_dynamic__0 = {
  fixtures: "# This is the title\\n\\nThis is the description\\n",
};
const a0 = "# This is the title\\n\\nThis is the description\\n";
const a1 = __raw_dynamic__0[fixtureDir];
const a2 = __raw_dynamic__1[fileName];

`;

exports[`macros 9. custom encoding: 9. custom encoding 1`] = `


import raw from '../macro';

const a0 = raw('./fixtures/markdown.md', 'utf-8');
const a1 = raw(\`./\${fixtureDir}/markdown.md\`, 'binary');
const a2 = raw(\`./fixtures/\${fileName}\`, 'binary');
      

      ↓ ↓ ↓ ↓ ↓ ↓

var __raw_dynamic__1 = {
  "javascript.js":
    'const sum = require("../sum");\\n\\ntest("add two numbers", () => {\\n  const result = sum(1, 2);\\n  expect(result).toEqual(3);\\n});\\n',
  "mark.md": "## Another markdown\\n",
  "markdown.md": "# This is the title\\n\\nThis is the description\\n",
};
var __raw_dynamic__0 = {
  fixtures: "# This is the title\\n\\nThis is the description\\n",
};
const a0 = "# This is the title\\n\\nThis is the description\\n";
const a1 = __raw_dynamic__0[fixtureDir];
const a2 = __raw_dynamic__1[fileName];

`;
